src/main/java/edu/kit/ipd/crowdcontrol/objectservice/quality/QualityIdentificator.java

```diff
@@ -0,0 +1,221 @@
+    private final Observable<Event<Rating>> ratingObservable = EventManager.RATINGS_CREATE.getObservable();
+   //TODO uncomment after exp.controller merge private final ExperimentController controller;
+    private final AnswerRatingOperations answerOperations;
+    private final ExperimentOperations experimentOperations;
+    private final AlgorithmOperations algorithmOperations;
+    private final Set<AnswerQualityStrategy> answerAlgorithms;
+    private final Set<RatingQualityStrategy> ratingAlgorithms;
+
+    private AnswerQualityStrategy answerIdentifier;
+    private RatingQualityStrategy ratingIdentifier;
+
+
+    /**
+     * Has to be  >= 0 and  <10
+     * <p>
+     * Might be set to allow more flexibility and more good answers
+     */
+    private int ratingQualityThreshold = 5;
+
+    public QualityIdentificator(AlgorithmOperations algorithmOperations, AnswerRatingOperations answerRatingOperations, ExperimentOperations experimentOperations) {
+
+        //TODO uncomment after contoller merge this.controller = controller;
+        this.answerOperations = answerRatingOperations;
+        this.experimentOperations = experimentOperations;
+        this.algorithmOperations = algorithmOperations;
+        this.answerAlgorithms = new HashSet<>();
+        this.ratingAlgorithms = new HashSet<>();
+
+        // Reflection isn't used, that's why algorithms need to be added manually
+        //------------------------------------------------------
+        //
+        //           ADD OTHER ALGORITHMS HERE
+        //
+        //------------------------------------------------------
+        answerAlgorithms.add(new AnswerQualityByRatings());
+        ratingAlgorithms.add(new RatingQualityByDistribution());
+
+
+
+        //Load algorithms in db
+        answerAlgorithms.forEach(algorithm -> algorithmOperations.storeAnswerQualityAlgorithm(new AlgorithmAnswerQualityRecord(algorithm.getAlgorithmName(),algorithm.getAlgorithmDescription()),algorithm.getParams()));
+        ratingAlgorithms.forEach(algorithm -> algorithmOperations.storeRatingQualityAlgorithm(new AlgorithmRatingQualityRecord(algorithm.getAlgorithmName(),algorithm.getAlgorithmDescription()),algorithm.getParams()));
+
+        ratingObservable.subscribe(rating -> this.onNext(rating.getData()));
+
+    }
+
+    /**
+     * This method is performed, if the RATINGS_CREATE-observable emits an event.
+     * All ratings and answers of the experiment will be rated. Furthermore the status of the experiment is checked and
+     * if the criteria are met, it will be shut-down
+     * @param rating which has been created an will be processed
+     */
+    private void onNext(Rating rating) {
+
+        Optional<ExperimentRecord> exp = experimentOperations.getExperiment(rating.getExperimentId());
+        if (!exp.isPresent()) {
+            throw new IllegalArgumentException("Error! Can't retrieve the experiment matching to ID:" + rating.getExperimentId());
+        }
```

Comment:
more beautiful: `experimentOperations.getExperiment(rating.getExperimentId()).orElseThrow(() -> new IllegalArgumentException(..))` then you don't have to call get() every time